streamer/periodconcat_node.py

```diff
@@ -0,0 +1,256 @@
+    for i, packager_node in enumerate(self._packager_nodes):
+      
+      dash_file_path = os.path.join(
+        packager_node._output_dir,
+        self._pipeline_config.dash_output)
+      
+      mpd = ElementTree.ElementTree(file=dash_file_path).getroot()
+      period = find(mpd, 'Period')
+      
+      period.attrib['id'] = str(i)
+      
+      if self._pipeline_config.streaming_mode == StreamingMode.VOD:
+        # For VOD, use the 'mediaPresentationDuration' to set the Period duration.
+        period.attrib['duration'] = mpd.attrib['mediaPresentationDuration']
+        total_time += ISO8601.parse(period.attrib['duration'])
+      else:
+        # For LIVE, set the start attribute.
+        period.attrib['start'] = str(total_time)
+        #total_time += ISO8601.parse(period.attrib['where to get the period time from???'])
+        
+      # For multi segment media
+      for segment_template in findall(
+        period,
+        'AdaptationSet',
+        'Representation',
+        'SegmentTemplate'):
+        path_to_segement = os.path.relpath(packager_node._output_dir, self._output_dir)
+        
+        if segment_template.attrib.get('initialization'):
+          segment_template.attrib['initialization'] = os.path.join(
+            path_to_segement,
+            segment_template.attrib['initialization']
+          )
+          
+        if segment_template.attrib.get('media'):
+          segment_template.attrib['media'] = os.path.join(
+            path_to_segement,
+            segment_template.attrib['media']
+          )
+      
+      # For single segment media
+      for base_url in findall(
+        period,
+        'AdaptationSet',
+        'Representation',
+        'BaseURL'):
+        path_to_segement = os.path.relpath(packager_node._output_dir, self._output_dir)
+        
+        assert base_url.text is not None, 'No media file found for the single segment ' + dash_file_path # For mypy's static analysis
+        base_url.text = os.path.join(
+          path_to_segement,
+          base_url.text
+        )
+        
+      periods.append(period)
+    
+    # Add the periods collected from all the files.
+    ref_mpd.extend(periods)
+    
+    # Update total duration for vod.
+    if self._pipeline_config.streaming_mode == StreamingMode.VOD:
+      ref_mpd.attrib['mediaPresentationDuration'] = str(total_time)
+    
+    # Remove the placeholder namespace put by the module and set it to be the default namespace.
+    ElementTree.register_namespace('', dash_namespace)
+    
+    
+    # Write the period concat to the outpud_dir.
+    with open(os.path.join(
+        self._output_dir,
+        self._pipeline_config.dash_output), 'w') as master_dash:
+      contents = "<?xml version='1.0' encoding='UTF-8'?>\n"
+      contents += "<!--Generated with https://github.com/google/shaka-packager -->\n"
+      contents += "<!--Made Multi-Period with https://github.com/google/shaka-streamer version {} -->\n".format(__version__)
```

Comment:
I love this so much, I wish it were there on single-period streams, too.

Since we're moving toward having Packager write directly to cloud storage, maybe we can add a feature to Packager where we can specify an additional comment to include.  Then we can ask Packager to add Streamer branding directly to all output files.